internal/clients/client.go

```diff
@@ -86,7 +91,16 @@ func (client *Client) Build(ctx context.Context, o *Option) error {
 		"api-version",
 		"$skipToken",
 	}
-
+	tok, err := o.Cred.GetToken(client.StopContext, policy.TokenRequestOptions{
```

Comment:
Is it possible that we move this part to the ResourceManagerAccount's lazy init function `GetObjectId()`?
Because requesting token here will slow down the azapi initialization.

Reply:
I have moved the token request to the GetObjectId() method. To do this I had to pass in the *Client as a dependency. Let me know what you think